packages/typespec-azure-resource-manager/lib/operations.tsp

```diff
@@ -694,3 +694,113 @@ op checkLocalNameAvailability<
   Response,
   AdditionalParams
 >;
+
+/**
+ * @dev An action performed at subscription level.
+ * @template Request The request model for the action.
+ * @template Response The response type for the action.
+ * @template Parameters Optional. Additional parameters after the path parameters (e.g. Location)
+ * @template Scope Optional. The scope of the action.
+ * @template Error Optional. The error response, if non-standard.
+ */
+@post
+@action
+@armResourceCollectionAction
+op ArmProviderActionSync<
+  Request extends {} | void = void,
+  Response extends {} | void = void,
+  Parameters extends {} = {},
+  Scope extends {} = SubscriptionActionScope,
+  Error extends {} = ErrorResponse
+>(
+  ...Azure.ResourceManager.Foundations.DefaultBaseParameters<Scope>,
```

Comment:
I think we should have one sync abd one async function implementation, and the two templates should `is` that function definition using different default parameters.  This way, changes in the implementation don't have to be repeated across all definitions.

Reply:
Mark, any suggestion about the file in where these functions should be (the sync and async)? operations.tsp is fine?